src/mmrelay/async_db_pool.py

```
@@ -0,0 +1,411 @@
+                if connection is None:
+                    logger.warning(
+                        "Async connection pool exhausted, waiting for available connection"
+                    )
+                    conn_id, connection = await self._wait_for_available_connection(
+                        max_wait_seconds=self.timeout
+                    )
+
+                    if connection is None:
+                        raise sqlite3.OperationalError(
+                            "Async connection pool exhausted and timeout reached"
+                        )
```

Ruff (0.14.3)

Abstract `raise` to an inner function

(TRY301)